src/H5Tinit_float.c

```diff
@@ -0,0 +1,552 @@
+            HGOTO_DONE(i);
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value);
+}
+
+/*-------------------------------------------------------------------------
+ * Function:    H5T__bit_cmp
+ *
+ * Purpose:     Compares two bit vectors and returns the index for the
+ *              first bit that differs between the two vectors.     The
+ *              size of the vector is NBYTES.  PERM is a mapping from
+ *              actual order to little endian.  Ignores differences where
+ *              the corresponding bit in pad_mask is set to 0.
+ *
+ * Return:      Index of first differing bit.
+ *-------------------------------------------------------------------------
+ */
+static unsigned int
+H5T__bit_cmp(unsigned int nbytes, int *perm, void *_a, void *_b, const unsigned char *pad_mask)
+{
+    unsigned int   i;
+    unsigned char *a = (unsigned char *)_a;
+    unsigned char *b = (unsigned char *)_b;
+    unsigned char  aa, bb;
+    unsigned int   ret_value = 0;
+
+    FUNC_ENTER_PACKAGE_NOERR
+
+    for (i = 0; i < nbytes; i++) {
+        assert(perm[i] < (int)nbytes);
+        if ((aa = (unsigned char)(a[perm[i]] & pad_mask[perm[i]])) !=
+            (bb = (unsigned char)(b[perm[i]] & pad_mask[perm[i]]))) {
+            unsigned int j;
+
+            for (j = 0; j < 8; j++, aa >>= 1, bb >>= 1) {
+                if ((aa & 1) != (bb & 1))
+                    HGOTO_DONE(i * 8 + j);
+            }
+        }
+    }
+
+done:
+    FUNC_LEAVE_NOAPI(ret_value);
+}
+
+/*-------------------------------------------------------------------------
+ * Function:    H5T__fix_order
+ *
+ * Purpose:     Given an array PERM with elements FIRST through LAST
+ *              initialized with zero origin byte numbers, this function
+ *              creates a permutation vector that maps the actual order
+ *              of a floating point number to little-endian.
+ *
+ *              This function assumes that the mantissa byte ordering
+ *              implies the total ordering.
+ *
+ * Return:      void
+ *-------------------------------------------------------------------------
+ */
+static void
+H5T__fix_order(int n, int last, int *perm, H5T_order_t *order)
+{
+    int i;
+
+    FUNC_ENTER_PACKAGE_NOERR
+
+    assert(last > 1);
+
+    /* We have at least three points to consider */
+    if (perm[last] < perm[last - 1] && perm[last - 1] < perm[last - 2]) {
+        /* Little endian */
+        *order = H5T_ORDER_LE;
+        for (i = 0; i < n; i++)
+            perm[i] = i;
+    }
+    else if (perm[last] > perm[last - 1] && perm[last - 1] > perm[last - 2]) {
+        /* Big endian */
+        *order = H5T_ORDER_BE;
+        for (i = 0; i < n; i++)
+            perm[i] = (n - 1) - i;
+    }
+    else {
+        /* Undetermined endianness - defaults to 'VAX' for historical
+         * reasons, but there are other mixed-endian systems (like ARM
+         * in rare cases)
+         */
+        assert(0 == n % 2);
+        *order = H5T_ORDER_VAX;
```

Comment:
Quincey's other comment implies this change drops VAX support, but this code appears to still support VAX (byte order at least). Does this code still do anything useful?

Reply:
Yes, no change in functionality, I was incorrect about my previous comment.

Reply:
Yes. I want to keep this in, even though VAX is defunct (we can change the H5T_ORDER_VAX name later). ARM allows mixed-endian floats and I would not be surprised to see other platforms do that as well in our heterogeneous future.